x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/append.test.tsx

```diff
@@ -0,0 +1,134 @@
+  tag: undefined,
+  ignore_failure: undefined,
+  description: undefined,
+};
+
+const APPEND_TYPE = 'append';
+
+describe('Processor: Append', () => {
+  let onUpdate: jest.Mock;
+  let testBed: SetupResult;
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(async () => {
+    onUpdate = jest.fn();
+
+    await act(async () => {
+      testBed = await setup({
+        value: {
+          processors: [],
+        },
+        onFlyoutOpen: jest.fn(),
+        onUpdate,
+      });
+    });
+    testBed.component.update();
+  });
+
+  test('prevents form submission if required fields are not provided', async () => {
+    const {
+      actions: { addProcessor, saveNewProcessor, addProcessorType },
+      form,
+    } = testBed;
+
+    // Open flyout to add new processor
+    addProcessor();
+    // Click submit button without entering any fields
+    await saveNewProcessor();
+
+    // Expect form error as a processor type is required
+    expect(form.getErrorsMessages()).toEqual(['A type is required.']);
+
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(APPEND_TYPE);
+
+    // Click submit button with only the type defined
+    await saveNewProcessor();
+
+    // Expect form error as "field" is required parameter
+    expect(form.getErrorsMessages()).toEqual([
+      'A field value is required.',
+      'A value is required.',
+    ]);
+  });
+
+  test('saves with default parameter values', async () => {
+    const {
+      actions: { addProcessor, saveNewProcessor, addProcessorType },
+      form,
+      find,
+    } = testBed;
+
+    // Open flyout to add new processor
+    addProcessor();
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(APPEND_TYPE);
+    // Add "field" value (required)
+    form.setInputValue('fieldNameField.input', 'field_1');
+
+    find('valueFieldAppend.input').simulate('change', [{ label: 'Some_Value' }]);
+
+    // form.setInputValue('valueField.comboBoxSearchInput', 'Some_Value');
+    // Save the field
+    await saveNewProcessor();
+
+    const processors = getProcessorValue(onUpdate, APPEND_TYPE);
+    expect(processors[0].append).toEqual({
+      field: 'field_1',
+      value: ['Some_Value'],
+      ...defaultAppendParameters,
+    });
+  });
+
+  test('allows optional parameters to be set', async () => {
+    const {
+      actions: { addProcessor, addProcessorType, saveNewProcessor },
+      form,
+      find,
+    } = testBed;
+
+    // Open flyout to add new processor
+    addProcessor();
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(APPEND_TYPE);
+    // Add "field" value (required)
+    form.setInputValue('fieldNameField.input', 'field_1');
+
+    // Set optional parameteres
+    find('valueFieldAppend.input').simulate('change', [{ label: ['Some_Value'] }]);
```

Comment:
I know this is just a test value, but can we replace `['Some_Value']` with just `'Some_Value'` as we don't expect this to ever be an array. Looking at line 131 it looks like we are returning an array of arrays which is not correct, I think.

Reply:
Can we not have multiple labels and stuff in a combo box? I thought that was why the array came back. @jloleysens

Reply:
OK I see it. It's fixed.